packages/dashboard/src/pages/Settings/components/dialogs/BackupDialog.tsx

```diff
@@ -1,56 +1,151 @@
-import ConfirmDialog from '../../../../components/ConfirmDialog'
-import { ChangeEvent, useState } from 'react'
-import { Checkbox, FormControlLabel } from '@material-ui/core'
-import { Services } from '../../../../API'
-import { useAsync } from 'react-use'
-import BackupPreviewCard from '../BackupPreviewCard'
+import { useContext, useMemo, useState, useEffect } from 'react'
+import { PluginServices, Services } from '../../../../API'
+import { useAsync, useAsyncFn } from 'react-use'
+import BackupContentSelector, { BackupContentCheckedStatus } from '../BackupContentSelector'
 import { useDashboardI18N } from '../../../../locales'
+import { MaskDialog, MaskTextField, useSnackbar } from '@masknet/theme'
+import { Box } from '@material-ui/core'
+import { UserContext } from '../../hooks/UserContext'
+import LoadingButton from '@material-ui/lab/LoadingButton'
+import { VerifyCodeRequest, fetchUploadLink, uploadBackupValue } from '../../api'
+import formatDateTime from 'date-fns/format'
+import { LoadingCard } from '../../../../components/Restore/steps/LoadingCard'
 
 export interface BackupDialogProps {
+    local?: boolean
+    params?: VerifyCodeRequest
     open: boolean
     onClose(): void
 }
 
-export default function BackupDialog({ open, onClose }: BackupDialogProps) {
+export default function BackupDialog({ local = true, params, open, onClose }: BackupDialogProps) {
+    const snackbar = useSnackbar()
     const t = useDashboardI18N()
-    const [checked, setChecked] = useState(true)
+    const [backupPassword, setBackupPassword] = useState('')
```

Comment:
plz use react-hook-form build the pr form

Reply:
I think this should be optional. We can use it in a complex form.